app/models/hyrax/work.rb

```diff
@@ -94,7 +94,7 @@ module Hyrax
   # @see https://wiki.lyrasis.org/display/samvera/Hydra::Works+Shared+Modeling
   #   for a historical perspective.
   class Work < Hyrax::Resource
-    include Hyrax::Schema(:core_metadata)
+    include  Hyrax::Schema(:core_metadata) unless Hyrax.config.flexible?
```

Comment:
I have two remarks pertaining to this (and all instances of `include Hyrax::Schema(:core_metadata)`) :⁠—

1. Could this be made more configurable (i·e, able to be disabled in general, not just when `Hyrax.config.flexible?`)? We use M3‐derived metadata already at Project Surfliner, and we’re not likely to switch over to using this method anytime soon, but being able to still disable `Hyrax::Schema(:core_metadata)` would be useful for us (the fact that it defines `:title` is a persistent source of annoyance.

2. `date_uploaded`, `date_modified`, and `depositor` are administrative metadata Hyrax needs to function. I’m not sure it is appropriate to be defining these in the same place as descriptive metadata. ① notwithstanding, I liked the separation that Hyrax enforced between core administrative metadata that the application required to work properly (`core_metadata`), and descriptive metadata which appears in the form and can be customized by developers (`basic_metadata`, etc).

Reply:
1. seems pretty viable to me. Flipping the flag from being "unless has flexible on" to "if autoload metadata" feels like a reasonable change in perspective and increases flexibility

2. I'd rather stay out of the "what metadata goes in which category" debate when it comes to the context fo this PR. The goal of this PR is to enable the fully M3 enabled and dynamically changeable structure, not to debate whether title belongs in core_metadata.

Reply:
My comment was less intended to litigate which metadata goes in which category and more to acknowledge that supporting multiple categories of metadata might be useful; i·e that maybe not every metadata attribute belongs in the same box. I don’t feel so strongly about this that I think it needs to hold up a merge, but I do think that it maybe should be discussed (with metadata folks) whether allowing/requiring metadata specialists to manage things like `date_uploaded` and `date_modified`, which do not appear in any of the web forms and cannot be deleted, is useful.

Along these lines, I notice that `Hyrax::AdministrativeSet` and `Hyrax::FileSet` are being managed with M3 in this MR. I definitely think it is good that they have that ability, but I also think it is entirely possible that people will _not_ want to have to manage AdminSet or FileSet classes as part of their M3 models. There’s currently no capacity for people to use M3 just for objects, or just for objects and collections, or just for descriptive metadata and not administrative metadata, and I think that those are useful constraints that people might want to enable.

Again, I don’t feel so strongly about this that I think it should hold up a merge, but I do think it is a conversation worth having with metadata experts, and a bit of proactive design towards enabling that kind of flexibility might be good.